src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -580,15 +579,10 @@ private static int GetIntQueryParameterValue(NameValueCollection queryStringName
                 IDurableOrchestrationClient client = this.GetClient(request);
 
                 object input = null;
-                if (request.Content != null)
+                if (request.Content?.Headers?.ContentLength > 0)

```

Comment:
The end result of 'request.Content?.Headers?.ContentLength' is `long?`, meaning we could still get a NullReferenceException here. You could fix this logic with `(request.Content?.Headers?.ContentLength  ?? 0 ) > 0`, though that is getting kind of ugly.

In general, I am not sure I like the new dependency on ContentLength. While almost all clients will appropriately assign a ContentLength header when sending a body, I am not sure if it is technically required in the HTTP spec. In the case where a client is forgetting to send the ContentLength header, we would now no longer reading their input. Instead, consider checking that `json` below is not null or empty.

Reply:
I don't think there is actually a risk of a null-ref here. Operators like `>` work fine with nullable value types. However, there is a risk with specifically checking for `ContentLength > 0`. I suggest instead that you compare `ContentLength != 0` to account for the missing `Content-Length` header.

Reply:
Huh, today I learned!

Reply:
Yeah I had to test it out when I was writing it because it looked odd. Most surprising was `request.Content?.Headers?.ContentLength == 0` is false when there's no value.